src/AWSS3.jl

```diff
@@ -471,52 +474,63 @@ end
 
 
 """
-    s3_list_objects([::AWSConfig], bucket, [path_prefix])
+    s3_list_objects([::AWSConfig], bucket, [path_prefix]; delimiter="/", max_items=1000)
 
 [List Objects](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGET.html)
 in `bucket` with optional `path_prefix`.
 
-Returns `Vector{Dict}` with keys `Key`, `LastModified`, `ETag`, `Size`,
+Returns an iterator of `Dict`s with keys `Key`, `LastModified`, `ETag`, `Size`,
 `Owner`, `StorageClass`.
 """
-function s3_list_objects(aws::AWSConfig, bucket, path_prefix="")
+function s3_list_objects(aws::AWSConfig, bucket, path_prefix=""; delimiter="/", max_items=nothing)
```

Comment:
Should the `delimiter` be a `Char`?

Reply:
I don't think so because we're passing it as a string to S3 anyways.